tests/Integration/ScriptsTest.php

```diff
@@ -334,131 +292,94 @@ public function test_user_logged_in(): void {
-		);
-
-		self::assertFalse(
-			wp_script_is( 'wp-parsely-tracker', 'enqueued' ),
-			'Failed to confirm tracker script was not enqueued'
-		);
-
-		wp_print_scripts();
-		$output = ob_get_clean();
-
-		self::assertSame(
-			'',
-			$output,
-			'Failed to confirm script tags were not printed'
+		// Current user is logged-in and track_authenticated_users is false so enqueue
+		// should fail. Confirm that tracker script is registered but not enqueued.
+		$this->assert_script_statuses(
+			'wp-parsely-tracker',
+			array( 'registered' ),
+			array( 'enqueued' )
 		);
 
+		// -- Test second blog.
 		// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.switch_to_blog_switch_to_blog
 		switch_to_blog( $second_blog );
 		TestCase::set_options( $custom_options );
+		$this->go_to_new_post();
 
+		// Check that we're on the second blog and that first user is not a member.
 		self::assertEquals( get_current_blog_id(), $second_blog );
-		self::assertFalse( is_user_member_of_blog( $new_user, get_current_blog_id() ) );
+		self::assertFalse( is_user_member_of_blog( $first_blog_admin, get_current_blog_id() ) );
 
-		ob_start();
+		// Enqueue tracker script.
 		self::$scripts->register_scripts();
 		self::$scripts->enqueue_js_tracker();
 
-		$intermediate_output = ob_get_contents();
-		self::assertSame(
-			'',
-			$intermediate_output,
-			'Failed to confirm scripts were not printed by enqueue_js_tracker()'
-		);
-
-		self::assertTrue(
-			wp_script_is( 'wp-parsely-tracker', 'enqueued' ),
-			'Failed to confirm tracker script was enqueued'
-		);
-
-		wp_print_scripts();
-		$output = ob_get_clean();
-
-		self::assertSame(
-			// phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedScript
-			"<script type='text/javascript' data-parsely-site=\"blog.parsely.com\" src='https://cdn.parsely.com/keys/blog.parsely.com/p.js?ver=" . Parsely::VERSION . "' id=\"parsely-cfg\"></script>\n",
-			$output,
-			'Failed to confirm script tags were printed correctly'
+		// First user is not logged-in to the second blog, so track_authenticated_users value
+		// is irrelevant. Confirm that tracker script is registered and enqueued.
+		$this->assert_script_statuses(
+			'wp-parsely-tracker',
+			array( 'enqueued', 'registered' )
 		);
 	}
 
 	/**
-	 * Test the wp_parsely_enqueue_js_tracker filter
-	 * When it returns false, the tracking script should not be enqueued.
+	 * Assert multiple enqueueing statuses for a script.
 	 *
-	 * @covers \Parsely\Scripts::enqueue_js_tracker
-	 * @uses \Parsely\Parsely::api_key_is_missing
-	 * @uses \Parsely\Parsely::api_key_is_set
-	 * @uses \Parsely\Parsely::get_options
-	 * @uses \Parsely\Parsely::post_has_trackable_status
-	 * @uses \Parsely\Parsely::update_metadata_endpoint
+	 * @param string $handle Script handle to test.
+	 * @param array  $assert_true Statuses that should assert to true.
+	 * @param array  $assert_false Statuses that should assert to false.
```

Comment:
```suggestion
	 * @param string $handle       Script handle to test.
	 * @param array  $assert_true  Optional. Statuses that should assert to true. Accepts 'enqueued', 'registered', 'queue', 'to_do', and 'done'. Default is an empty array.
	 * @param array  $assert_false Optional. Statuses that should assert to false. Accepts 'enqueued', 'registered', 'queue', 'to_do', and 'done'. Default is an empty array.
```
Do we need to make this method slightly more protected, so that someone doesn't try to take advantage of the array args being optional and simply call `assert__script_statuses( 'foo' )` and get a pass? 

(This may already be handled via no assertions being called, therefore it is marked as [Risky](https://phpunit.readthedocs.io/en/9.5/risky-tests.html#useless-tests) automatically.)

Reply:
We could do this. Would you suggest a particular approach to this?

Reply:
If the test is run without the args, does it get flagged as Risky? If so, no change is needed, since I think we have that configured as a fail (if not, we should).

Reply:
Passing no arguments does flag a test as risky. However, this is not the case if we have multiple calls to `assert_script_statuses` and at least one of them has arguments, as the test does assert something in that case.

Reply:
Good point - in which case, having a check inside `assert_script_statuses()` itself and mark test as risky / incomplete seems sensible then.

Reply:
Done